src/Cassandra.IntegrationTests/TestClusterManagement/SCassandraManager.cs

```diff
@@ -0,0 +1,184 @@
+    {
+
+        private static Process _scassandraProcess;
+
+        public static void Start()
+        {
+            Stop();
+            _scassandraProcess = new Process();
+            var jarPath = Environment.GetEnvironmentVariable("SCASSANDRA_JAR");
+            if (string.IsNullOrEmpty(jarPath))
+            {
+                throw new Exception("SCassandra: SCASSANDRA_JAR environment variable not set!");
+            }
+            _scassandraProcess.StartInfo.FileName = "java";
+            _scassandraProcess.StartInfo.Arguments = string.Format("-jar {0}", jarPath);
+            _scassandraProcess.StartInfo.UseShellExecute = false;
+            _scassandraProcess.StartInfo.CreateNoWindow = true;
+            _scassandraProcess.StartInfo.RedirectStandardOutput = true;
+            _scassandraProcess.StartInfo.RedirectStandardError = true;
+#if !NETCORE
+            _scassandraProcess.StartInfo.WindowStyle = ProcessWindowStyle.Hidden;
+#endif
+            var eventWaitHandler = new AutoResetEvent(false);
+            var primingPortReady = false;
+            var scassandraPortReady = false;
+            var isReady = false;
+            var isPortAlreadyUsed = false;
+            _scassandraProcess.OutputDataReceived += (sender, e) =>
+            {
+                if (e.Data == null || isReady) return;
+                if (e.Data.Contains("Port 8042 ready for Cassandra binary connections"))
+                {
+                    scassandraPortReady = true;
+                }
+                if (e.Data.Contains("Bound to localhost/127.0.0.1:8043"))
+                {
+                    primingPortReady = true;
+                }
+
+                if (e.Data.Contains("Unable to bind to port 8042"))
+                {
+                    isPortAlreadyUsed = true;
+                    eventWaitHandler.Set();
+                }
+
+                if (scassandraPortReady && primingPortReady)
+                {
+                    isReady = true;
+                    eventWaitHandler.Set();
+                }
+            };
+            _scassandraProcess.ErrorDataReceived += (sender, e) =>
+            {
+                if (e.Data == null) return;
+                Console.WriteLine(e.Data);
+            };
+            _scassandraProcess.Start();
+
+            _scassandraProcess.BeginOutputReadLine();
+            _scassandraProcess.BeginErrorReadLine();
+
+            eventWaitHandler.WaitOne(20000);
+            if (isPortAlreadyUsed)
+            {
+                Stop();
+                throw new Exception("Scassandra not started, port 8042 already in use");
+            }
+            if (!isReady)
+            {
+                Stop();
+                throw new Exception("SCassandra not started!");
+            }
+        }
+
+        public static void Stop()
+        {
+            if (_scassandraProcess == null) return;
+
+            try
+            {
+                _scassandraProcess.Kill();
```

Comment:
Instead of using `Kill` and `WaitForExit()`, I think it should be `process.Dispose()`.